content/spin/kv-store.md

```diff
@@ -0,0 +1,405 @@
+title = "Persistent Data: Spin"
+template = "common_main"
+date = "2023-02-21T00:00:00Z"
+enable_shortcodes = true
+[extra]
+
+---
+- [Key Value Storage With Spin Applications](#key-value-storage-with-spin-applications)
+- [SQLite](#sqlite)
+  - [Redis, PostgreSQL \& SQLite](#redis-postgresql--sqlite)
+- [Tutorial Prerequisites](#tutorial-prerequisites)
+- [Creating a New Application](#creating-a-new-application)
+- [Configuration](#configuration)
+  - [The Spin TOML File](#the-spin-toml-file)
+- [Using the Spin SDK](#using-the-spin-sdk)
+  - [The Spin SDK Version](#the-spin-sdk-version)
+- [Building and Deploying Your Spin Application](#building-and-deploying-your-spin-application)
+- [Storing and Retrieving Data From Your Default Key/Value Store](#storing-and-retrieving-data-from-your-default-keyvalue-store)
+- [Conclusion](#conclusion)
+- [Next Steps](#next-steps)
+
+## Key Value Storage With Spin Applications
+
+Spin applications are best suited for event-driven, stateless workloads that have low-latency requirements. Keeping track of the application's state (storing information) is an integral part of any useful product or service. For example, users (and the business) will expect to store and load data/information at all times; during an application’s execution. In [Spin v0.9.0](https://www.fermyon.com/blog/spin-v09), we've added support for key/value storage. This is a storage option for applications that need data in the form of key/value pairs and are satisfied by a BASE consistency model. Workload examples include general value caching, session caching, counters, and serialized application state.
+
+## SQLite
+
+SQLite is an integral part of Spin's key/value API. As of Spin v0.9.0 onwards, users can easily access, a built-in, _local_ key/value SQLite database in every Spin application. This persistent storage feature is available by default and with minimal configuration. Supported language SDKs currently include Go, JavaScript/TypeScript, and Rust. With these new changes implemented.
+
+> Spin users can now persist and retrieve non-relational data from a key/value store across multiple requests to and from the same application; written in any of these languages.
+
+### Redis, PostgreSQL & SQLite
```

Comment:
I am curious if this section overemphasizes embedded SQLite over the key/value feature. Right now SQLite is ONE host implementation of the API. This may be a good opportunity to mention that more host implementations can be added so the same spin app can target other databases such as redis.